draft/ZEP0004.md

```diff
@@ -0,0 +1,194 @@
+
+It seems likely that some features that are first supported as conventions may be promoted to extensions if they are deemed sufficiently general.
+
+Conventions can also help users switch between different storage libraries more flexibly.
+Since Zarr and HDF5 implement nearly identical data models, a single convention can be applied to both formats.
+This allows downstream software to maintain better separation of concerns between storage and domain-specific logic.
+
+## Usage and Impact
+
+We demonstrate the usage and impact with a simple example: unit encoding.
+
+A convention around unit encoding might look like this
+
+> **Convention Name**: "units-v1"
+>
+> Information about the units of a Zarr array should be stored in the `units` attribute and encoded as strings using
+> the [MIXF-08](http://people.csail.mit.edu/jaffer/MIXF/MIXF-08) standard.
+
+To create data with this convention using zarr-python, one would write
+
+```python
+import zarr
+z = zarr.open(
+    'example_with_units.zarr', mode='w', shape=(10000, 10000), chunks=(1000, 1000), dtype='f4'
+)
+z.attrs['units'] = 'm^2' 
+z.attrs['zarr_convention'] = "units-v1"
+```
+
+Reading back the data with zarr-python would work just fine. However, reading the data with a unit-aware package would enable automatic decoding.
+This is possible today with [Pint](https://pint.readthedocs.io/), [Xarray](https://xarray.pydata.org/), and [Pint-Xarray](https://pint-xarray.readthedocs.io/)
+
+```python
+import xarray as xr
+import pint_xarray
+ds = xr.open_dataset('example_with_units.zarr', engine='zarr').pint.quantify()
+```
+
+Now the variables have been decoded to unit-aware Pint arrays, based on the unit information encoded via the conventions.
+Likewise, we can serialize the data using the convention back to Zarr automatically.
+
+```python
+ds.dequantify().to_zarr("array_with_units.zarr")
+```
+
+None of this requires any awareness of units on part of Zarr itself.
+
+## Backward Compatibility
+
+Conventions by definition are fully backwards and forwards compatible with all versions of Zarr, including both V2 and V3.
+They require no changes to the spec or extensions and can be ignored by Zarr implementations.
+
+## Detailed description
+
+This ZEP itself describes the _process_ by which conventions may be proposed, discussed, accepted, and published by the Zarr community.
+This process is intended to be much more lightweight and informal than a spec change, which requires a ZEP.
+
+If this ZEP is accepted, Conventions will be added to the <https://zarr-specs.readthedocs.io/> website as a new top-level heading
+and a corresponding folder will be created in the `zarr-developers/zarr-specs` repo.
+A conventions template, which accompanies this ZEP, will be added to that folder.
+
+### New Convention Process
+
+New conventions are proposed via a pull-request to the `zarr-specs` repo which adds a new conventions document.
```

Comment:
Not clear what the pull-request contains (a domain specific convention document ?). If it contains a convention document, a template should describe its structure ?

Reply:
Yes, a template is needed. I need to finish this PR.